content/spin/serverless-ai-api-guide.md

```diff
@@ -0,0 +1,157 @@
+title = "Serverless API"
+template = "spin_main"
+date = "2023-09-05T09:00:00Z"
+enable_shortcodes = true
+[extra]
+url = "https://github.com/fermyon/developer/blob/main/content/spin/serverless-ai-api-guide.md"
+
+---
+- [Using Fermyon Serverless AI From Applications](#using-fermyon-serverless-ai-from-applications)
+
+Spin provides an interface for you to perform AI inference within Spin applications. By default, a given component of an app will not have access to any Serverless AI models. Access must be provided explicitly via the component manifest.  For example, a component could be given access to the llama2-chat model using:
+
+```toml
+[[component]]
+ai_models = ["llama2-chat"]
+```
+
+> Note: To deploy a Spin application (that performs AI inferencing) to Fermyon Cloud please visit the [Serverless AI](https://developer.fermyon.com/cloud/serverless-ai#accessing-private-beta) section in the documentation. It covers signing up for the private beta and setting up your Fermyon Serverless AI.
+
+## Using Fermyon Serverless AI From Applications
+
+The Spin SDK surfaces the Serverless AI interface to your language. The set of operations is common across all supporting SDKs:
+
+| Operation  | Parameters | Returns | Behavior |
+|------------|------------|---------|----------|
+| `infer`  | model `string`, prompt `string`, params `list` | `string`  | The `infer` is performed on a specific model. The name of the model is the first parameter provided (i.e. `llama2-v70b-chat`, `llama2-v13b-chat`, `llama2-v7b-chat` or other; passed in as a `string`). The second parameter is a prompt; passed in as a `string`. The third parameter is a mix of float and unsigned integers relating to inferencing parameters in this order: `max-tokens` (unsigned 32 integer), `repeat-penalty` (float 32), `repeat-penalty-last-n-token-count` (unsigned 32 integer), `temperature` (float 32), `top-k` (unsigned 32 integer) and `top-p` (float 32). The result from `infer` is a `string` |
```

Comment:
The model names are incorrect:

```suggestion
| `infer`  | model `string`, prompt `string`, params `list` | `string`  | The `infer` is performed on a specific model. The name of the model is the first parameter provided (i.e. `llama2-chat`, `codellama-instruct`, or other; passed in as a `string`). The second parameter is a prompt; passed in as a `string`. The third parameter is a mix of float and unsigned integers relating to inferencing parameters in this order: `max-tokens` (unsigned 32 integer), `repeat-penalty` (float 32), `repeat-penalty-last-n-token-count` (unsigned 32 integer), `temperature` (float 32), `top-k` (unsigned 32 integer) and `top-p` (float 32). The result from `infer` is a `string` |
```

Also, the Rust and JS SDK don't take strings but rather take in some sort of enum. This gets translated to the string that ultimately gets passed over the wit interface as a string. This is just a convenience so that users don't have to know a set of magic strings.